src/charwise.rs

```diff
@@ -863,6 +863,179 @@ impl Serializable for State {
 mod tests {
     use super::*;
 
+    #[test]
+    fn test_double_array() {
+        /*
+         *          a--> 4
+         *         /
+         *   a--> 1 --b--> 5
+         *  /
+         * 0 --c--> 3 --b--> 6
+         *  \
+         *   b--> 2
+         *
+         *   a = Ａ
+         *   b = Ｃ
+         *   c = Ｂ
```

Comment:
```suggestion
         *          Ａ--> 4
         *         /
         *   Ａ--> 1 --Ｃ--> 5
         *  /
         * 0 --Ｂ--> 3 --Ｃ--> 6
         *  \
         *   Ｃ--> 2
         *
         *   Ａ= 0
         *   Ｃ= 1
         *   Ｂ= 2
```
I felt it was difficult to understand the correspondence between `a,b,c` and `A,B,C`.
How about the new one?

Reply:
Thanks